File: tests/algorithms/x/PHP/conversions/octal_to_hexadecimal.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _intdiv($a, $b) {
    if (function_exists('bcdiv')) {
        $sa = is_int($a) ? strval($a) : (is_string($a) ? $a : sprintf('%.0f', $a));
        $sb = is_int($b) ? strval($b) : (is_string($b) ? $b : sprintf('%.0f', $b));
        return intval(bcdiv($sa, $sb, 0));
    }
    return intdiv($a, $b);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function octal_to_hex($octal) {
  global $nums, $t, $num;
  $s = $octal;
  if (strlen($s) >= 2 && substr($s, 0, 0 + 1 - 0) == '0' && substr($s, 1, 1 + 1 - 1) == 'o') {
  $s = substr($s, 2, strlen($s) - 2);
}
  if (strlen($s) == 0) {
  $panic('Empty string was passed to the function');
}
  $j = 0;
  while ($j < strlen($s)) {
  $c = substr($s, $j, $j + 1 - $j);
  if ($c != '0' && $c != '1' && $c != '2' && $c != '3' && $c != '4' && $c != '5' && $c != '6' && $c != '7') {
  $panic('Not a Valid Octal Number');
}
  $j = $j + 1;
};
  $decimal = 0;
  $k = 0;
  while ($k < strlen($s)) {
  $d = ord(substr($s, $k, $k + 1 - $k));
  $decimal = $decimal * 8 + $d;
  $k = $k + 1;
};
  $hex_chars = '0123456789ABCDEF';
  if ($decimal == 0) {
  return '0x';
}
  $hex = '';
  while ($decimal > 0) {
  $idx = $decimal % 16;
  $hex = substr($hex_chars, $idx, $idx + 1 - $idx) . $hex;
  $decimal = _intdiv($decimal, 16);
};
  return '0x' . $hex;
};
  $nums = ['030', '100', '247', '235', '007'];
  $t = 0;
  while ($t < count($nums)) {
  $num = $nums[$t];
  echo rtrim(octal_to_hex($num)), PHP_EOL;
  $t = $t + 1;
}
$__end = _now();
$__end_mem = memory_get_peak_usage();
$__duration = max(1, intdiv($__end - $__start, 1000));
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;

File: tests/algorithms/transpiler/FS/conversions/octal_to_hexadecimal.fs
// Generated 2025-08-07 10:31 +0700

exception Return
let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _substring (s:string) (start:int) (finish:int) =
    let len = String.length s
    let mutable st = if start < 0 then len + start else start
    let mutable en = if finish < 0 then len + finish else finish
    if st < 0 then st <- 0
    if st > len then st <- len
    if en > len then en <- len
    if st > en then st <- en
    s.Substring(st, en - st)

let _idx (arr:'a array) (i:int) : 'a =
    if i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
let __bench_start = _now()
let __mem_start = System.GC.GetTotalMemory(true)
let rec octal_to_hex (octal: string) =
    let mutable __ret : string = Unchecked.defaultof<string>
    let mutable octal = octal
    try
        let mutable s: string = octal
        if (((String.length (s)) >= 2) && ((string (s.[0])) = "0")) && ((string (s.[1])) = "o") then
            s <- _substring s 2 (String.length (s))
        if (String.length (s)) = 0 then
            failwith ("Empty string was passed to the function")
        let mutable j: int = 0
        while j < (String.length (s)) do
            let c: string = string (s.[j])
            if (((((((c <> "0") && (c <> "1")) && (c <> "2")) && (c <> "3")) && (c <> "4")) && (c <> "5")) && (c <> "6")) && (c <> "7") then
                failwith ("Not a Valid Octal Number")
            j <- j + 1
        let mutable decimal: int = 0
        let mutable k: int = 0
        while k < (String.length (s)) do
            let d: int = unbox<int> (string (s.[k]))
            decimal <- (decimal * 8) + d
            k <- k + 1
        let hex_chars: string = "0123456789ABCDEF"
        if decimal = 0 then
            __ret <- "0x"
            raise Return
        let mutable hex: string = ""
        while decimal > 0 do
            let idx: int = ((decimal % 16 + 16) % 16)
            hex <- (string (hex_chars.[idx])) + hex
            decimal <- decimal / 16
        __ret <- "0x" + hex
        raise Return
        __ret
    with
        | Return -> __ret
let nums: string array = [|"030"; "100"; "247"; "235"; "007"|]
let mutable t: int = 0
while t < (Seq.length (nums)) do
    let num: string = _idx nums (t)
    printfn "%s" (octal_to_hex (num))
    t <- t + 1
let __bench_end = _now()
let __mem_end = System.GC.GetTotalMemory(true)
printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)
